pkg/resource/deploy/providers/doc.go

```diff
@@ -0,0 +1,16 @@
+// Copyright 2016-2021, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package providers contains the logic for first-class providers (versioning, loading, resource operations).
```

Comment:
Nit: this package manages _all_ providers: the core engine mostly doesn't distinguish between default and non-default providers.

```suggestion
// Package providers contains the logic for managing provider resources (versioning, loading, resource operations).
```